### PR TITLE
chore: to wrap intlMessage in a div in ContentNotification

### PR DESCRIPTION
**Pull request by YahiaElTai**
This PR wraps IntlMessqge in a div in ContentNotification to have a `block` layout for intl message, whereas for children the layout will stay as-is `flex`.
This PR is a result of this issue #1476

### CODE REVIEW

File: packages/components/notifications/src/content-notification/README.md [hunk 1]
```diff
@@ -16,10 +16,11 @@ import { ContentNotification } from '@commercetools-uikit/notifications';
...
-| `type`        | `string`       |    ✅    | `success`, `info`, `warning`, `error` | -       | Type of notification, determining its appearance.                    |
-| `children`    | `node`         | ✅ (\*)  | -                                     | -       | Message to be shown inside a notification panel.                     |
-| `intlMessage` | `intl message` | ✅ (\*)  | -                                     | -       | An intl message object that will be rendered with `FormattedMessage` |
+| Props           | Type           | Required | Values                                | Default | Description                                                          |
+| --------------- | -------------- | :------: | ------------------------------------- | ------- | -------------------------------------------------------------------- |
+| `type`          | `string`       |    ✅    | `success`, `info`, `warning`, `error` | -       | Type of notification, determining its appearance.                    |
+| `containerType` | `string`       |          | `flex`, `block`                       | `flex`  | Type of container for children, determining its layout.              |
```
Comment (tdeekens) [inline packages/components/notifications/src/content-notification/README.md:22]: MDN calls this a "display type". The CSS spec seems to differentiate between `display-inside` and `display-outside` values. `block` being a `display-outside` value and `flex` a `display-inside` value.

That brings me to two ideas:

1. `displayType` might be a better name
2. Are we just wanting to control row or column layout in the end?

I wonder if anybody ever assumed that you can layout two children inside the notification. So changing anything here isn't much breaking more of a bug fix and using `flex` and direction of sorts would be fine? Sorry for discussing it here again.

Comment (YahiaElTai) [reply packages/components/notifications/src/content-notification/README.md:22]: from my side, `displayType` sounds more appropriate and we are trying to control only the row layout.

**Comment by YahiaElTai** [comment]
> Hmm shouldn't we simply wrap the `FormattedMessage` or `children` into a `div`?
> The current flex container can stay as it is, as long as the container rendering the message is not a flex box container...right? 🤔

This is the current solution we have on the MC but we thought maybe this should be also fixed on UiKit. fix is the wrong word here maybe make the component a bit more flexible and remove the need to add `divs` around

**Comment by tdeekens** [comment]
> Hmm shouldn't we simply wrap the FormattedMessage or children into a div?

That's what I wonder now. We're trying to work around a breaking change. That's cool. However, I wonder if the current behaviour is not rather a bug. How intends to layout two children in a block? Maybe I am missing something 🦆.

**Comment by emmenko** [comment]
My point was to question if we need this new prop instead of just "fixing" the rendered markup, for example by wrapping the rendered children into a `div`.

**Comment by YahiaElTai** [comment]
> > Hmm shouldn't we simply wrap the FormattedMessage or children into a div?
> 
> That's what I wonder now. We're trying to work around a breaking change. That's cool. However, I wonder if the current behaviour is not rather a bug. How intends to layout two children in a block? Maybe I am missing something 🦆.

It's definitely not a bug, but more like asking too much of the component to handle our use cases. as we said before maybe the component shouldn't enforce layouts on its text children but since this is the case, the only options now are to either wrap a div around the children or add a new prop here, I'm fine with both as long as we agree on one solution.

**Comment by YahiaElTai** [comment]
```javascript
<ContentNotification type="error">
  <FormattedMessage
     {...messages[resource]}
     values={{ limit, customersupport: getCustomerSupportLink }}
   >
      {(message) => <div>{message}</div>}
   </FormattedMessage>
</ContentNotification>
```

This was actually my solution to handle this case on the MC.

**Comment by emmenko** [comment]
What I'm suggesting is to wrap this ⬇️ into a `div`

https://github.com/commercetools/ui-kit/blob/5734a2507bb7d5bc1a1eb32aab166aff43e8fc32/packages/components/notifications/src/content-notification/content-notification.js#L133-L137

Which is the same essentially of doing:

```js
<ContentNotification type="error">
  <div>
    Hello there
  </div>
</ContentNotification>
```

but it's done in the component itself, which also covers the case when passing the `intlMessage` prop.

**Comment by YahiaElTai** [comment]
> What I'm suggesting is to wrap this ⬇️ into a `div`
> 
> https://github.com/commercetools/ui-kit/blob/5734a2507bb7d5bc1a1eb32aab166aff43e8fc32/packages/components/notifications/src/content-notification/content-notification.js#L133-L137
> 
> Which is the same essentially of doing:
> 
> ```js
> <ContentNotification type="error">
>   <div>
>     Hello there
>   </div>
> </ContentNotification>
> ```
> 
> but it's done in the component itself, which also covers the case when passing the `intlMessage` prop.

Yeah I think this is a good solution and works well for us. if everyone agrees on the solution, I will close the PR 🚀  @tdeekens @jonnybel

**Comment by emmenko** [comment]
PS: You can do the changes in the PR, no need to open a new one.

**Comment by YahiaElTai** [comment]
> PS: You can do the changes in the PR, no need to open a new one.

ah sorry I misunderstood your solution, if we add a `div` into the component itself that will be a breaking change because it will change the layout from `flex` to `block`, I thought you were speaking about adding a div into the MC when using the component as a consumer.

**Comment by emmenko** [comment]
Well yes and no. I think what we can do at least is to wrap the rendered `FormattedMessage` in a `<div>`, when the `intlMessage` prop is provided. I think this can be considered a **fix**.

As for the `children`, it could potentially be a breaking change yes.
It's very unlikely that someone used two children, thus having a flex behavior.
We can either:
- keep it as-is, then consumers should pass the `<div>` on their own (if necessary).
- wrap it also in a `<div>`, and sell it as bug fix.

So I would suggest maybe to do as following:
- for the `intlMessage` case, we wrap it in a `<div>`.
- for the `children` case, we keep it as-is, but we document the need to pass a `<div>` in case the flex behavior is unwanted.
- add a TODO to fix the wrapping container in the next major release.

**Comment by tdeekens** [comment]
Thanks @emmenko that's what I felt a bit too. That we sell it more as a bug fix as it feels odd to rely on the current layout which feels mostly broken. I think keeping the `children` case as is is good. Maybe when @YahiaElTai adopts this in the MC he can already check if/how many cases we have if any where we rely on it.

File: .changeset/pretty-onions-punch.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-uikit/notifications': patch
+---
+
+chore: wrap intlMessage prop in a div to have a block layout instead of flex
```
Comment (emmenko) [inline .changeset/pretty-onions-punch.md:5]:
```suggestion
fix: wrap `intlMessage` prop in a `<div>` to have a block layout instead of flex
```

**Pull request merged**